lib/jekyll/commands/serve.rb

```diff
@@ -172,31 +283,51 @@ def enable_logging(opts)
         # forget to add one of the certificates.
 
         private
-        # rubocop:disable Metrics/AbcSize
         def enable_ssl(opts)
-          return if !opts[:JekyllOptions]["ssl_cert"] && !opts[:JekyllOptions]["ssl_key"]
-          if !opts[:JekyllOptions]["ssl_cert"] || !opts[:JekyllOptions]["ssl_key"]
+          jekyll_opts = opts[:JekyllOptions]
+          return if !jekyll_opts["ssl_cert"] && !jekyll_opts["ssl_key"]
+          if !jekyll_opts["ssl_cert"] || !jekyll_opts["ssl_key"]
             # rubocop:disable Style/RedundantException
             raise RuntimeError, "--ssl-cert or --ssl-key missing."
           end
+
+          Jekyll.logger.info "LiveReload:", "Serving over SSL/TLS.  If you are using a "\
+            "certificate signed by an unknown CA, you will need to add an exception "\
+            "for both #{jekyll_opts["host"]}:#{jekyll_opts["port"]} and "\
+            "#{jekyll_opts["host"]}:#{jekyll_opts["livereload_port"]}"
+
           require "openssl"
           require "webrick/https"
-          source_key = Jekyll.sanitized_path(opts[:JekyllOptions]["source"], \
-                    opts[:JekyllOptions]["ssl_key" ])
-          source_certificate = Jekyll.sanitized_path(opts[:JekyllOptions]["source"], \
-                    opts[:JekyllOptions]["ssl_cert"])
+          source_key = Jekyll.sanitized_path(
+            jekyll_opts["source"],
+            jekyll_opts["ssl_key"]
+          )
+          source_certificate = Jekyll.sanitized_path(
+            jekyll_opts["source"],
+            jekyll_opts["ssl_cert"]
+          )
           opts[:SSLCertificate] =
             OpenSSL::X509::Certificate.new(File.read(source_certificate))
           opts[:SSLPrivateKey ] = OpenSSL::PKey::RSA.new(File.read(source_key))
           opts[:SSLEnable] = true
         end
 
         private
-
         def start_callback(detached)
           unless detached
             proc do
-              Jekyll.logger.info("Server running...", "press ctrl-c to stop.")
+              @running << "."
```

Comment:
I feel that `Queue` is the wrong data type for this usage. Correct me if I'm wrong, but you never seem to actually wait for the `Queue` to become populated and even if you did, there are probably better data types for this usage.